rasterio/rio/merge.py

```diff
@@ -36,31 +36,39 @@ def merge(ctx, files, driver):
             with rasterio.open(files[0]) as first:
                 kwargs = first.meta
                 kwargs['transform'] = kwargs.pop('affine')
-                dest = np.empty((first.count,) + first.shape, 
+                dest = np.zeros((first.count,) + first.shape, 
                     dtype=first.dtypes[0])
 
+            nodataval = 0.0
```

Comment:
@sgillies shouldn't this be based on the data type for ```dest``` or that automatically handled?

This could be:
```nodataval = dest.flat[0]```